sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -616,18 +618,22 @@ pi_result piDeviceGetInfo(pi_device device, pi_device_info paramName,
       return static_cast<pi_result>(ret_err);
 
     result = supported;
-    std::memcpy(paramValue, &result, sizeof(cl_bool));
+    std::memcpy(paramValue, &result, sizeof(bool));
```

Comment:
```suggestion
    std::memcpy(paramValue, &result, sizeof(result));
```

? To make it future-proof in case we decide to change return data type for this query

Reply:
I think in general the PI opencl backend has a problem with not respecting the size requested by the `paramSize` argument. I hope UR will amend this.